src/write.rs

```diff
@@ -755,13 +755,28 @@ impl<S: Span> Report<'_, S> {
             }
 
             // Note
-            if let (Some(note), true) = (&self.note, is_final_group) {
-                if !self.config.compact {
-                    write_margin(&mut w, 0, false, false, true, Some((0, false)), &[], &None)?;
-                    write!(w, "\n")?;
+            if is_final_group {
+                for (i, note) in self.notes.iter().enumerate() {
+                    if !self.config.compact {
+                        write_margin(&mut w, 0, false, false, true, Some((0, false)), &[], &None)?;
+                        writeln!(w)?;
+                    }
+                    let note_prefix = format!("{} {}", "Note", i + 1);
```

Comment:
A little troubled by the lack of customisation in the message here. I think it's better to have the consumer of the API specify these things by not adding the extra `Note <number>`.

Reply:
I don't see it part of the PR tbh, as you can't change "Note" in singular form either. So it would be a breaking change...

Reply:
I think other programs like rustc only print "note" over and over, and IMO that's acceptable behaviour. What do you think?

Reply:
It goves a reference point on which note it is and how many there are.

@zesterer what do you think? Keep or not to keep the number?

Reply:
I think this is fine on reflection, I managed to misremember how my own crate worked beforehand!